extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -180,8 +187,8 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
 
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                return ((getActiveConnections(a).size() + 1) * b.getConnectionWeight() -
```

Comment:
Okay, that makes sense.  So I guess what we're going for here is keeping the ratio under the specified limit - that is, if 10:1 is the configured ratio (based on configured weights of the servers), then work up to that, then drop back down.  It's still, mathematically, unbalanced (you end up with a 5.5:1 ratio), but it keeps it "in bounds" so to speak.

I've tweaked the code such that it computes the net weight of each of the connections, and, if they're equal, returns the difference between the configured weight of each system, otherwise returns the difference in the net weight.  Does that look more like what we're going for in a WLC algorithm?